scripts/monitor/visualization/resource_monitoring.py

```diff
@@ -0,0 +1,596 @@
+from typing import Dict
+import matplotlib.pyplot as plt
+from bokeh.models import Title
+from matplotlib.backends.backend_pdf import PdfPages
+import numpy
+import pandas as pd
+import pandas_bokeh
+from operator import truediv
+
+
+class ResourceUsage:
+    instances = 0  # count number of class instances (can't find a way to update this when an instance is deleted)
+
+    def __init__(self, meta_runtime: pd.DataFrame, metrics: pd.DataFrame) -> None:
+        """
+        initialization step: bring meta, runtime, and metrics data into scope
+        :param meta_runtime: meta_runtime table
+        :param metrics: metrics table
+        """
+        self.meta_runtime = meta_runtime
+        self.metrics = metrics
+        self.instances += 1
+
+    def run(self) -> None:  # maybe do this as part of initialization
+        """
+        Do everything that needs to be done to set up the webpage, and then set up the webpage
+        :return: None
+        """
+        self.generate_plot_files()
+        self.generate_report_homepage()
+
+    def generate_report_homepage(self) -> None:
+        """
+        Set up webpage
+        :return: None
+        """
+
+        pass
+        # jekyll stuff
+
+    def generate_plot_files(self) -> None:
+        """
+        Create all files and put them where other functions can find them
+        :return: None
+        """
+        tasks = self.num_task_shards()
+        unsharded_tasks = dict([(k, v) for k, v in tasks.items() if v == 0])
+        sharded_tasks = dict([(k, v) for k, v in tasks.items() if v != 0])
+        # unsharded_tasks = dict(filter(lambda elem: elem[1] == 0, tasks.items()))  # pycharm false positive?
+        # sharded_tasks = dict(filter(lambda elem: elem[1] != 0, tasks.items()))
+        for task in unsharded_tasks:
```

Comment:
do you mean to iterate through the items?

Reply:
yeah I could probably have done this with a list comprehension instead of using multiple lines